packages/docs/Examples/sale.md

```diff
@@ -5,120 +5,244 @@ icon: flame
 
-Essentially Seller is saying: you can do with this inputs-output pair whatever you want as long as you spend the output indicated in the first input and you keep the locking script and the amount of the first output the same. Seller can confidentially publish such a partially signed transaction as he wants to sell the nft spent by the fist inputs for the amount indicated in the first output and both are signed.
+Seller calls `encode` with an environment `{ n: ..., p: ... }`, indicating that the first inputs will spend `n` and the second inputs will spend `p`. Seller will use the expression `${Sale} Sale.exec(n, p)`. As the `exec` function of the `Sale` contract returns an array `[p, n]` the first output wil represent `p` and the second output will represent `n`. This is exactly the "crossover" transaction described above.
+
+```ts
+export class Sale extends Contract {
+  static exec(n: NFT, p: Payment) {
+    const [ownerN] = n._owners
+    const [ownerP] = p._owners
+    n.transfer(ownerP)
+    p.transfer(ownerN)
+    return [p, n]
+  }
+}
+```
 
-#### Partially Signed Transactions
+The first argument to the `exec` function is an nft of the following class.
 
-Todo.
+```ts
+export class NFT extends Contract {
+  constructor(name = '', symbol = '') {
+    super({ name, symbol })
+  }
 
-### Helper Classes
+  transfer(to: string) {
+    this._owners = [to]
+  }
+}
+```
 
-Buyer will use the smart contract below to create a smart object holding the satoshis to pay for the NFT.
+The second argument is created by Buyer to pay for the nft.
 
 ```ts
 class Payment extends Contract {
-  constructor(owner: string, _amount: number) {
-    super({ _owners: [owner], _amount })
+  constructor(_amount: number) {
+    super({ _amount })
   }
 
   transfer(to: string) {
     this._owners = [to]
   }
+
+  setAmount(a: number) {
+    this._amount = a
+  }
 }
 ```
 
-Seller will use this helper class to mock up the input that will later be used to spend Buyers's payment object.
+In the following sections, we explain the process of seller minting an NFT and creating a sales transaction as well as a buyer going through a purchase.
+
+### Minting an NFT
+
+To mint an nft, Seller can use the `computer.new` function.
 
 ```ts
-const randomPublicKey = '023e21361b53bb2e625cc1f41d18b35ae882e88d8d107df1c3711fa8bc54db8fed'
-const mockedRev = 'mock:0000000000000000000000000000000000000000000000000000000000000000:0'
+const nft = await seller.new(NFT, ['name', 'symbol'])
+```
+
+### Building the Sales Transaction
+
+The challenge with building the sales transaction is that the output that contains the payment is not yet known when the transaction is built. This is a problem because to evaluate an expression on the Bitcoin Computer either a smart object or the revision of an existing smart object is needed for all parameters of the call.
```

Comment:
The challenge in creating the sales transaction arises from the fact that the output containing the payment is not determined at the time of building the transaction. This presents a problem because, in order to evaluate an expression on the Bitcoin Computer, either a smart object or a revision of an existing smart object is required for all parameters of the call.